server/app/repository/VersionRepository.java

```diff
@@ -507,6 +510,13 @@ private void removeCacheForVersion(String versionKey) {
     }
   }
 
+  private void removeCacheForProgram(String programKey) {
```

Comment:
**suggestion (non-blocking):** `uncacheProgram`? idk

Reply:
We also have removeCacheForVersion, so I feel like if we change this method name we should change that one too to uncacheVersion. I'll do both of these in a separate PR.